### PR TITLE
Initial support for openSUSE (step 1)

### PR DESCRIPTION
**Pull request by Gsantomaggio**
## Proposed Changes
Add Support for [openSUSE](https://www.opensuse.org/) linux distribution 
## Types of Changes

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [ ] All tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

## Further Comments
This is a Work in Progress branch, with the current chef version, we can't use openSUSE leap 15 because of https://github.com/chef/ohai/issues/1343

Even less relevant, there is also this issue https://github.com/chef/bento/issues/1185  with Vagrant

I am using Leap 42 fot the tests, when https://github.com/chef/ohai/issues/1343 will be fixed I will switch to Leap 15.

There are a few points to fix, so this branch may not work correctly.

### CODE REVIEW

File: resources/erlang_zypper_repository_on_suse.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+#
+# Cookbook Name:: rabbitmq
+# Resource:: erlang_zypper_repository_on_suse
```
Comment (michaelklishin) [inline resources/erlang_zypper_repository_on_suse.rb]: The naming we use for these resources is `erlang_[type]_repository_on_[hosting service]` :)

Comment (michaelklishin) [inline resources/erlang_zypper_repository_on_suse.rb]: Never mind my earlier comment. I suggest that we rename this to `erlang_zypper_repository_on_suse_factory` or even just `erlang_zypper_repository`? For other RPMs I can imagine alternative locations such as GitHub or may even PackageCloud one day (they have their own Chef cookbook, though).

Would it make sense to expect more than one package location or be extra specific that we use SUSE Factory here?

Comment (Gsantomaggio) [reply resources/erlang_zypper_repository_on_suse.rb]: Will rename in `erlang_zypper_repository_on_suse_factory`.
We have to use the Factory repositories in Leap distros to use le last RabbitMQ and Erlang versions

**Comment by Gsantomaggio** [comment]
- solved the `gpg key` problem by adding `gpgautoimportkeys`.
- add start epmd service 

The `converge`  seems to [work correctly](https://paste.opensuse.org/view//82562574) :
```
sudo  KITCHEN_LOCAL_YAML=.kitchen.dokken.yml kitchen converge  rabbitmq-erlang-latest-suse-opensuse-leap-42
....
Running handlers:
Running handlers complete
Chef Client finished, 8/21 resources updated in 21 seconds
       Finished converging <rabbitmq-erlang-latest-suse-opensuse-leap-42> (0m23.10s).
-----> Kitchen is finished. (0m25.69s)
```

We already updated our SUSE RabbitMQ setup to [3.7.14](https://build.opensuse.org/request/show/689898)

I think there is still work to do, but at least the `converge` should be ok :)

**Comment by michaelklishin** [comment]
@Gsantomaggio I assume this requires Chef/Ohai master and thus too early to QA?

**Comment by Gsantomaggio** [comment]
No,  `Ohai: 14.8.10`. 

I am using leap 43.2 for the tests. It that does not have [this problem](https://github.com/chef/ohai/issues/1343) because it uses the [`/etc/SuSE-release`](https://github.com/chef/ohai/blob/master/lib/ohai/plugins/linux/platform.rb#L226) to decode the `platform_family`. 

`SuSE-release` was removed in Leap 15* this is why stopped working.

**Comment by Gsantomaggio** [comment]
>thus too early to QA?

So we have 2 ways:
1- wait the fix for leap 15, 
2- release for the leap 42.3 (end of the life June 30 2019) to test everything and complete the task. When 1 is ready move to it.

**Comment by michaelklishin** [comment]
I see, let's conitnue with 42.3 for now. I'll get to it this week.

**Comment by michaelklishin** [comment]
Merged manually with some corrections. Once I figure out how to provision Chef 15 on OpenSUSE I'll continue with Leap 15.1 since `rabbitmq-server-plugins` for 3.7.x doesn't seem to be available in earlier releases.

This is definitely a good start. Thank you, @Gsantomaggio!

**Comment by michaelklishin** [comment]
Looks like [Chef Omnibus installer](https://omnitruck.chef.io/install.sh) has the same problem as in https://github.com/chef/ohai/pull/1299. So key pieces of the Chef ecosystem lacking support for OpenSUSE Leap 15.1 or Tumbleweeed 😞

**Pull request closed without merging**